### DISCUSSION TITLE
如何解决externalGlobals的包名包含`-`

### DISCUSSION THREAD

**Discussion by 1shin-7** [lisonge/vite-plugin-monkey#122] [Q&A] [answered]
如题，当externalGlobals的包名包含`-` 时，产物内的包名也会跟随这个名称从而导致改变原意：
![image](https://github.com/lisonge/vite-plugin-monkey/assets/49867372/08a72c4f-8f1f-44c5-956a-2a5f602e5fdb)
![image](https://github.com/lisonge/vite-plugin-monkey/assets/49867372/e06ec723-8cfe-427b-a343-ebc329724cdb)
可以发现这样出来的产物会使得浏览器认为是进行减法运算 ` wrtie - excel - file`

**Comment by lisonge** [comment, marked as answer]
因为你填错了，那个地方应该填 iife/umd 格式的导出变量标识符

**Reply by 1shin-7** [reply]
唔...请问可以指导一下如何修改嘛

**Reply by lisonge** [reply]
<https://unpkg.com/write-excel-file@1.4.27/bundle/write-excel-file.min.js>

查看这个文件，可以看出它是一个 umd 格式的文件，导出变量标识符是 `writeXlsxFile`

**Reply by 1shin-7** [reply]
解决了，谢谢大佬解答！之前对cdn.* 理解有误，以为第一个参数对应的是name，刚才仔细看了下才发现是exportVarName :joy:
